src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -134,28 +150,47 @@ where
 /// Propagates errors from convert shares
 /// # Panics
 /// Propagates panics from convert shares
-pub async fn convert_all_bits<F, C, S>(
+pub async fn convert_all_bits<'a, F, C, S, I>(
     ctx: &C,
-    locally_converted_bits: &[Vec<BitConversionTriple<S>>],
-) -> Result<Vec<Vec<S>>, Error>
+    mut locally_converted_bits: I,
+    num_bits: u32,
+    num_multi_bits: u32,
+) -> Result<impl Iterator<Item = Vec<Vec<S>>> + 'a, Error>
 where
     F: Field,
     C: Context<F, Share = S>,
-    S: ArithmeticSecretSharing<F>,
+    S: ArithmeticSecretSharing<F> + 'a,
+    I: Iterator<Item = Vec<BitConversionTriple<S>>>,
 {
-    let futures = locally_converted_bits
-        .iter()
+    let num_records = locally_converted_bits.size_hint().0;
+
+    let count = ((num_bits + num_multi_bits - 1) / num_multi_bits) as usize;
```

Comment:
I'll admit - I have no idea where this equation came from or what it does.

Reply:
This rounds `num_bits` up to the next multiple of `num_multi_bits`.  I've not looked at why...